engine/src/main/java/com/odysseusinc/arachne/executionengine/execution/r/DockerOverseer.java

```diff
@@ -58,17 +59,36 @@ public DockerOverseer(
         }));
         pos = stdout.length();
         this.client = client;
-        init.thenAccept(containerId ->
-                executor.scheduleWithFixedDelay(this::writeLogs, updateInterval, updateInterval, TimeUnit.MILLISECONDS)
-        );
+        init.thenAccept(containerId -> {
+            executor.scheduleWithFixedDelay(this::writeLogs, updateInterval, updateInterval, TimeUnit.MILLISECONDS);
+        });
         this.init = init;
     }
 
     @Override
     public CompletableFuture<ExecutionOutcome> abort() {
-        return result.isDone() ? result : CompletableFuture.completedFuture(
-                new ExecutionOutcome(Stage.ABORT, "Abort is not yet supported for docker analysis", null)
-        );
+        if (!init.isDone()) {
+            return CompletableFuture.completedFuture(new ExecutionOutcome(Stage.ABORTED, null, "Container initialization was canceled before it started."));
```

Comment:
This method still does nothing when init is not done. It does not prevent container from being started, e.g. the original defect https://github.com/OHDSI/ArachneExecutionEngine/pull/371#issuecomment-1894968994 still stands

Reply:
The first check ensures that if `init `is not done, it immediately returns a completed `ExecutionOutcome `indicating that container initialization was canceled. This prevents the Docker container from being started if initialization hasn't completed. Then, we first check if the container is still running using the `isContainerRunning()` method. If the container is running, we attempt to stop it; otherwise, we return a message indicating that the container has already completed. This ensures that you don't attempt to stop a container that has already finished its execution.

Reply:
"This prevents the Docker container from being started if initialization hasn't completed." - I don't see how can this be true. In my understanding `init` spins up docker container. I see no code here that will interfere with that.

Reply:
As our conversation on Friday, we are now using `CompletableFuture.cancel()` to stop the underlying process if it's already started, this will attempt to cancel the future but also, we are ensuring that a running process is stopped.